content/articles/2017-08-purrr-0.2.3.Rmd

```diff
@@ -0,0 +1,176 @@
+The plucking mechanism used for indexing into data structures with
+`map()` has been extracted into the function `pluck()`. Plucking is
+often more readable to extract an element buried in a deep data
+structure. Compare this syntax-heavy extraction which reads
+non-linearly:
+
+```{r, eval = FALSE}
+accessor(x[[1]])$foo
+```
+
+to the equivalent pluck:
+
+```{r, eval = FALSE}
+x %>% pluck(1, accessor, "foo")
+```
+
+The new function `attr_getter()` generates attribute accessors that
+can be used in `pluck()`:
+
+```{r}
+dfs <- list(iris, mtcars)
+dfs %>% pluck(2, attr_getter("row.names"))
+```
+
+or in mapped indexing:
+
+```{r}
+x <- list(
+  list(vec = letters, df = mtcars),
+  list(vec = LETTERS, df = iris)
+)
+x %>% map(list("df", attr_getter("row.names")))
+```
+
+
+### Useful tweaks and features
+
+* `set_names()` (now reexported from rlang) can now take a function to
+  tranform the names programmatically, and you can supply names in
+  `...` to reduce typing even more more.
+
+* You can now refer to arguments by position in lambda-formulas. The
+  `..1` symbol refers to the first argument, `..2` to the second and
+  so on. This makes it easier to use functions like `pmap()` with the
+  formula shortcut.
+
+* All typed map functions now treat `NULL` the same way as an empty
+  vector: they return an empty vector if any input is empty.
+
+* `auto_browse(f)` returns a new function that automatically calls
+  `browser()` if `f` throws an error.
+
+* `vec_depth()` computes the depth (i.e. the number of levels of
+  indexing) or a vector.
+
+* `reduce2()` and `reduce2_right()` make it possible to reduce with a
+  3 argument function where the first argument is the accumulated
+  value, the second argument is `.x`, and the third argument is `.y`.
+
+* `list_modify()` extends `stats::modifyList()` to replace by position
+  if the list is not named. `list_merge()` operates similarly to
+  `list_modify()` but combines instead of replacing.
+
+* `modify_depth()` (previously `at_depth()`) gains new `.ragged`
+  argument, and negative depths are now computed relative to the
+  deepest component of the list.
+
+
+### API Changes
+
+purrr no longer depends on lazyeval or Rcpp (or dplyr, as of the
+previous version). This makes the dependency graph of the tidyverse
+simpler, and makes purrr more suitable as a dependency of lower-level
+packages.
+
+A number of functions have been renamed or deprecated:
+
+* `is_numeric()` and `is_scalar_numeric()` are deprecated because they
+  don't test for what you might expect at first sight.
+
+* `as_function()` is now `as_mapper()` because it is a tranformation
+  that makes sense primarily for mapping functions, not in general.
+  The rlang package has an `as_function()` coercer that is smaller in
+  scope.
+
+* The data frame suffix `_df` has been (soft) deprecated in favour of
+  `_dfr` to more clearly indicate that it's a row-bind. All variants
+  now also have a `_dfc` for column binding. (These will not be
```

Comment:
I think you can remove the parenthetical since doesn't `bind_rows()` and `bind_cols()` have better behaviour now? Maybe it should be moved up a heading and get a small example?